File: test/fixtures/sample.kt
/**
 * Kotlin Sample File for Method Extraction Testing
 */

class Calculator(private var value: Int = 0) {

    fun add(x: Int, y: Int): Int {
        return x + y
    }

    private fun subtract(x: Int, y: Int): Int {
        return x - y
    }

    suspend fun asyncOperation(): Int {
        delay(100)
        return value
    }

    inline fun inlineFunction(block: () -> Unit) {
        block()
    }

    companion object {
        fun multiply(x: Int, y: Int): Int {
            return x * y
        }
    }
}

fun standaloneFunction(param: Int): Int {
    return param * 2
}

suspend fun asyncStandaloneFunction(): String {
    return fetchData()
}

fun <T> genericFunction(item: T): T {
    return item
}

fun String.extensionFunction(): Int {
    return this.length
}

data class DataClass(val name: String, val age: Int)

File: test/fixtures/sample.cpp
/**
 * C++ Sample File for Method Extraction Testing
 */

#include <iostream>
#include <string>

class Calculator {
private:
    int value;

public:
    Calculator(int initialValue = 0) : value(initialValue) {}

    int add(int x, int y) {
        return x + y;
    }

    int subtract(int x, int y) const {
        return x - y;
    }

    static int multiply(int x, int y) {
        return x * y;
    }

    virtual int compute() {
        return value;
    }

    inline int getValue() const {
        return value;
    }
};

class AdvancedCalculator : public Calculator {
public:
    AdvancedCalculator() : Calculator(0) {}

    int compute() override final {
        return getValue() * 2;
    }

    virtual void reset() {
        // Reset implementation
    }
};

int standaloneFunction(int param) {
    return param * 2;
}

template<typename T>
T genericFunction(T item) {
    return item;
}

namespace Math {
    int power(int base, int exp) {
        int result = 1;
        for(int i = 0; i < exp; i++) {
            result *= base;
        }
        return result;
    }
}
